include/hipSYCL/sycl/libkernel/detail/fp16/bitcasts.h

```diff
@@ -1,7 +1,7 @@
 #pragma once
-#include "hipSYCL/sycl/libkernel/host/host_backend.hpp"
 #ifndef FP16_BITCASTS_H
 #define FP16_BITCASTS_H
+#include "hipSYCL/sycl/libkernel/host/host_backend.hpp"
```

Comment:
Looking at this again, I don't think this should be included here - in fact, I wonder if it might be responsible for messing up compilation with nvc++ (`libkernel/backend.hpp` should already include host backend). Could this be a preexisting bug in this header?

See the comment here:
https://github.com/OpenSYCL/OpenSYCL/blob/7cff8d4b8ca5d818d67b6fc6a23b9f14b1302e01/include/hipSYCL/sycl/libkernel/backend.hpp#L35

Reply:
Hmm.. I think the root cause for that ending up in there is the fact that we have a circular dependency here, where sscp_backend.hpp includes the (fp16) builtins and the builtins reference the backend again, but never reaching the definitions of the `HIPSYCL_UNIVERSAL_TARGET` macro.
```
.. include/hipSYCL/sycl/libkernel/backend.hpp
... include/hipSYCL/sycl/libkernel/cuda/cuda_backend.hpp
... include/hipSYCL/sycl/libkernel/hip/hip_backend.hpp
... include/hipSYCL/sycl/libkernel/spirv/spirv_backend.hpp
... include/hipSYCL/sycl/libkernel/sscp/sscp_backend.hpp
.... include/hipSYCL/glue/llvm-sscp/ir_constants.hpp
..... include/hipSYCL/glue/llvm-sscp/s1_ir_constants.hpp
..... include/hipSYCL/glue/llvm-sscp/s2_ir_constants.hpp
.... include/hipSYCL/sycl/libkernel/sscp/builtins/core.hpp
..... include/hipSYCL/sycl/libkernel/sscp/builtins/builtin_config.hpp
...... include/hipSYCL/sycl/libkernel/sscp/builtins/../../memory.hpp
...... include/hipSYCL/sycl/libkernel/detail/half_representation.hpp
....... include/hipSYCL/sycl/libkernel/backend.hpp
....... include/hipSYCL/sycl/libkernel/detail/bit_cast.hpp
....... include/hipSYCL/sycl/libkernel/detail/int_types.hpp
....... include/hipSYCL/sycl/libkernel/detail/fp16/fp16.h
........ include/hipSYCL/sycl/libkernel/detail/fp16/bitcasts.h
......... include/hipSYCL/sycl/libkernel/backend.hpp
```

That being said, I "fixed this" by just moving the includes after the macros in `sscp_backend.hpp`. I think the other changes are still applicable.